packages/compartment-mapper/src/parse-cjs-shared-export-wrapper.js

```diff
@@ -131,7 +140,22 @@ export const wrap = (moduleEnvironmentRecord, compartment, resolvedImports) => {
     } else {
       return namespace;
     }
-  });
+  };
+  if (readPowers && readPowers.requireResolve) {
+    require.resolve = freeze((specifier, options) =>
+      readPowers.requireResolve(location, specifier, options),
```

Comment:
Now all that's left to do is convince Typescript I can call this function if I check the field exists even thought it's optional on readPowers

Reply:
it stopped complaining when I did this
```js
 if (typeof readPowers === 'object' && readPowers.requireResolve) {
    const { requireResolve } = readPowers;
    require.resolve = freeze((specifier, options) =>
      requireResolve(location, specifier, options),
    );
```
I don't appreciate what TS is trying to do for me here